frontend/src/components/misc/KowlJsonView.tsx

```diff
@@ -10,143 +10,33 @@
-    if (timerId != undefined) clearTimeout(timerId);
-
-    timerId = setTimeout(() => {
-        timerId = undefined;
-        action();
-    }, duration);
-};
-
-// Used for  Tooltip modifiers
-type TooltipPopperRect = { x: number; y: number; width: number; height: number };
-
-export const KowlJsonView = observer((props: ReactJsonViewProps) => {
-    const { style, ...restProps } = props;
-    const toast = useToast()
-
-    const { isOpen, onOpen, onClose } = useDisclosure();
-
-    const settings = uiSettings.jsonViewer;
-    const mergedStyles = Object.assign({ fontSize: settings.fontSize, lineHeight: settings.lineHeight, whiteSpace: 'normal' }, style);
-
-    const containerRef = React.useRef<HTMLDivElement>(null);
-
-    useEffect(() => {
-        // set keyup/keydown event listeners
-        const handleKeyboard = (e: KeyboardEvent) => (ctrlDown = e.ctrlKey);
-        document.addEventListener('keydown', handleKeyboard);
-        document.addEventListener('keyup', handleKeyboard);
-
-        return () => {
-            // clean up keyup/keydown event listeners
-            document.removeEventListener('keydown', handleKeyboard);
-            document.removeEventListener('keyup', handleKeyboard);
-        };
-    }, []);
-
-    return (
-        <div
-            className="copyHintContainer"
-            ref={containerRef}
-            onMouseDownCapture={e => {
-                if (e.ctrlKey) {
-                    // copy
-                    return;
-                }
-
-                // Show hint
-
-                // Find parent "div.variable-value"
-                // can be either the actual target (numbers), or a parent (strings, but only sometimes)
-                let target = e.target as Element;
-                if (target instanceof Element) {
-                    while (!target.classList.contains('variable-value')) {
-                        if (target.parentElement != null) target = target.parentElement; // try parent
-                        else return; // no more parents, give up
-                    }
-                }
-
-                const parentPos = containerRef.current ? containerRef.current.getClientRects()[0] : { x: 0, y: 0 };
-                clickPos.x = e.clientX - parentPos.x;
-                clickPos.y = e.clientY - parentPos.y;
-
-                onClose();
-                setOrRefreshTimeout(1000, () => {
-                    onClose();
-                });
-                onOpen();
-            }}
+import { CSSProperties } from 'react';
+import { Box } from '@redpanda-data/ui';
+import KowlEditor from './KowlEditor';
+
+export const KowlJsonView = observer((props: {
+    srcObj: object | string | null | undefined,
+    style?: CSSProperties,
+}) => {
+
+    const str = typeof props.srcObj == 'string'
+        ? props.srcObj
+        : JSON.stringify(props.srcObj, undefined, 4);
+
+    return <>
+        <Box minHeight="150px" maxHeight="600px"
```

Comment:
nit: maybe it would be nice to use Chakra values instead of px for consistency (and vertical rhytm in the future)

Reply:
addressed in 4bf7af9